mono/utils/mono-mmap.h

```diff
@@ -65,6 +65,12 @@ MONO_API int   mono_file_unmap_fileio (void *addr, void *handle);
 #endif
 MONO_API int   mono_mprotect   (void *addr, size_t length, int flags);
 
+// Last two parameters are optional.
+// This is mono_file_map but with optionally returning an error message.
+// See https://github.com/mono/mono/issues/8225.
+MONO_API void*
```

Comment:
I think, I thought -- MONO_API is needed here for the mkbundle output to use it. I know there was a PR to change that binding though. Let me see if it went through -- it'd be good if this failed to build, i.e. if I could only call the runtime through the new binding and not in the old way.

Reply:
Where is it used in mkbundle?

Reply:
I thought it was but indeed maybe not. I need to really build/test mkbundle more, on Windows. I haven't managed to do that yet.

Reply:
Isn't this MONO_API needed for driver to link dynamically to mono?
I could, granted, move the entire function into driver.
I was also torn on the char** error vs. a MonoError*.
As well, if you just want to take the alignment fix and say there are no errors to report for the time being, that a bit subjective but not complete nonsense -- code will eventually fail, and code needs error reporting, but it is a matter of degree. The error was plain to see with a little debugging, arguably.

Reply:
`MONO_API` is not needed for `driver.c` since it's statically linked. Since the function didn't exist before, and is not in a public header, it is not possible for it to be externally accessed (even with backward compatibility) cannot be used by any existing tool.

Remove the `MONO_API`, we will add it **if and only if** it's required by any external tool that we see fit.

Reply:
I understand  export only if necessary but it didn't look statically linked.
I'll try.

```
mono_boehm_SOURCES = \
	main.c

mono_boehm_LDADD = \
	$(MONO_LIB)		\
	$(glib_libs)		\
	$(LLVMMONOF)		\
	-lm			\
	$(MONO_DTRACE_OBJECT)

if STATIC_MONO
# Link libmono into mono statically
# This leads to higher performance, especially with TLS
MONO_LIB=$(boehm_static_libraries)
MONO_SGEN_LIB=$(sgen_static_libraries)
else 
MONO_LIB=libmonoboehm-2.0.la
MONO_SGEN_LIB=libmonosgen-2.0.la
endif
```